src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1890,6 +1898,39 @@ private function registerLockConfiguration(array $config, ContainerBuilder $cont
         }
     }
 
+    private function registerSemaphoreConfiguration(array $config, ContainerBuilder $container, PhpFileLoader $loader)
+    {
+        $loader->load('semaphore.php');
+
+        foreach ($config['resources'] as $resourceName => $resourceStore) {
+            $storeDsn = $container->resolveEnvPlaceholders($resourceStore, null, $usedEnvs);
+            $storeDefinition = new Definition(SemaphoreStoreInterface::class);
+            $storeDefinition->setFactory([SemaphoreStoreFactory::class, 'createStore']);
+            $storeDefinition->setArguments([$storeDsn]);
```

Comment:
should this pass $resourceStore instead?

Reply:
this is identical to lock configuration: https://github.com/symfony/symfony/blob/b240404d62909805d0c78816c35659d6daef4ba7/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L1867-L1870

Reply:
:shrug: I miss a piece of the puzzle.. in my case, in both cases, the dumped container contains
```
$container->privates['lock.default.factory'] = $instance = new \Symfony\Component\Lock\LockFactory(
  \Symfony\Component\Lock\Store\StoreFactory::createStore(
    $container->getEnv('LOCK_DSN')
  )
);
```

Reply:
Lock fixed in https://github.com/symfony/symfony/pull/46013